agents-for-bedrock/use-case-examples/cost-explorer-agent/agent_template.yaml

```diff
@@ -0,0 +1,396 @@
+
+  AgentFoundationalModel:
+    Type: String
+    Description: "Foundational model of the Bedrock agent"
+    Default: "anthropic.claude-3-haiku-20240307-v1:0"
+    AllowedValues:
+      - "anthropic.claude-3-haiku-20240307-v1:0"
+      - "anthropic.claude-3-sonnet-20240229-v1:0"
+
+  AgentInstruction:
+    Type: String
+    Description: "Instruction for the Bedrock agent"
+    Default: |
+      You are an expert in AWS, particularly in AWS services, billing, FinOps (Financial Operations), and analyzing cost and usage data. Your primary role is to assist users in understanding their AWS cost and usage trends, simplifying complex information into easily understandable summaries. Provide answers to any questions users have regarding their AWS services and AWS costs and usage. Additionally, you'll examine the gathered data on AWS costs and usage to provide recommendations, emphasizing strategies for cost efficiency and optimal resource utilization. Moreover, provide clear explanations and definitions for terms like AWS usage type, operation, and other aspects related to AWS cost and billing.
+  EnableSlackIntegration:
+    Type: String
+    Description: "Enable Slack integration"
+    Default: "false"
+    AllowedValues:
+      - "true"
+      - "false"
+
+  SlackBotToken:
+    Type: String
+    Description: "Slack Bot Token for the Slack integration"
+    NoEcho: true
+
+  SlackSigningSecret:
+    Type: String
+    Description: "Slack Signing Secret for the Slack integration"
+    NoEcho: true
+
+  SourceBucket:
+    Type: String
+    Description: "Name of the S3 bucket containing source files"
+Conditions:
+  CreateSlackIntegration: !Equals 
+    - !Ref EnableSlackIntegration
+    - "true"
+Resources:
+  # IAM role for the Lambda function that handles AWS billing data
+  BillingAgentLambdaRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      RoleName: !Sub "${AgentName}-lambda-role-${AWS::Region}"
+      ManagedPolicyArns:
+        - "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Action: "sts:AssumeRole"
+            Principal:
+              Service: "lambda.amazonaws.com"
+      Policies:
+        - PolicyName: LambdaCostExplorerPolicy
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "ce:GetDimensionValues"
+                  - "ce:GetCostAndUsage"
+                  - "ce:GetTags"
+                Resource: "*"
+
+  # Lambda function that handles AWS billing data
+  BillingAgentLambdaFunction:
+    Type: "AWS::Lambda::Function"
+    Properties:
+      FunctionName: !Sub "${AgentName}-${AWS::Region}"
+      Runtime: "python3.12"
+      Timeout: 300
+      MemorySize: 1024
+      Layers:
+        - !Sub "arn:aws:lambda:${AWS::Region}:336392948345:layer:AWSSDKPandas-Python312:13"
```

Comment:
hardcoded account number here.. you should provide the lambda layer as well

Reply:
This is a layer provided by AWS - updated the hardcoded account id with relevant SSM parameter